Common/TypeList.h

```diff
@@ -85,7 +85,7 @@ namespace typelist
  *     Tail (second element, can be another typelist)
  */
 template <typename H, typename T>
-struct TypeList
+struct ITK_TEMPLATE_EXPORT TypeList
```

Comment:
I wonder if `ITK_TEMPLATE_EXPORT` is really useful for the templates in Common/TypeList.h, as those templates do not have any member functions.

Reply:
I think the ITK_TEMPLATE_EXPORT is only needed for structs and classes that have `virtual` member functions (the types that may be used for `dynamic_cast`). So they should not be necessary for the templates in Common/TypeList.h